server/app/routes.py

```diff
@@ -13,3 +19,20 @@ def videos_raw():
 	@app.route('/videos')
 	def videos():
 		return jsonify(rss.summarize_watchlist(app.config['watchlist']))
+
+	@app.route('/watchlist', methods=['POST', 'DELETE'])
+	def modify_watchlist():
+		# only required value is 'id'
+		json = request.get_json()
+		if 'id' not in json:
+			return 'invalid json'
+		args = {}
+		_copy_values(args, json, ('id', 'source', 'site', 'name'))
+
+		watchlist = Watchlist(app.config['watchlist'])
+		if request.method == 'POST':
+			watchlist.add_channel(**args)
+		if request.method == 'DELETE':
+			watchlist.remove_channel(**args)
+		watchlist.write()
+		return 'success'
```

Comment:
this isn't conventional http behavior. We should either return data (sending back the feed for the channel would be nice), or nothing (with 200 response code).

Reply:
done